### DISCUSSION TITLE
Customizing the location of page chunks

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [vuepress/core#567] [Q&A] [answered]
Hello!

I'm just getting started with Vuepress and am evaluating if this is going to be a good fit for my use case. The initial impressions are promising, but I've a concern. I am trying to work on a protected and private documentation website. The contents of it should not be available without authentication.

I am going to gate access with Netlify's Role Based Access Control. I've set up the roles and redirects for server-side auth. That's working fine. To prevent client-side routing, I'm using NavigationGuards along with a Vuex store (it simply stores if the user is authenticated). However, there remains a last part of the problem. The chunks exported by Webpack have the content of the Markdown files pretty easily visible. For example, I got this as the export:

![image](https://user-images.githubusercontent.com/11403551/146156676-24116e3d-f48a-40e1-bb76-cb295ba3529c.png)

This is one of the files in `./dist/assets/js/v-[hash].js`. I am assuming all the page-based chunks are going to be exported with the `v-` prefix. So, could I configure a different directory for these? Maybe `/js/chunks/`? So, I can add these files to the "gated" content as well. If I gate the entire website, the basic files required for the functioning of the website also don't load which is expected.

So, if the above is not possible, what do all the fellow developers recommend when trying to setup a sensitive documentation? Looking forward to some great advice.

P.S.: Please ignore `hello-vitepress` text. I was trying out Vitepress before, but switched to Vuepress.

**Comment by Mister-Hope** [comment, marked as answer]
Sorry for the late reply, but if you want acess controll, you should never consider "A static site generator", you probably need to choose those with backend. Those tool may give you opportunity to dynamicly render markdown content, and they are surely more secure than tools like vitepress or vuepress.

Also people may only concern which folder does all the static files been generated (to avoid conflict), instead of their structure and file names, so your request is probably more like an edge case, and may out of our consideration.
